src/palace/manager/sqlalchemy/model/identifier.py

```diff
@@ -932,26 +985,32 @@ def classify(
                 identifier=self,
                 subject=subject,
                 data_source=data_source,
+                create_method_kwargs={"weight": weight},
             )
         except MultipleResultsFound as e:
             # TODO: This is a hack.
-            all_classifications = _db.query(Classification).filter(
+            all_classifications_query = _db.query(Classification).filter(
                 Classification.identifier == self,
                 Classification.subject == subject,
                 Classification.data_source == data_source,
             )
-            all_classifications = all_classifications.all()
+            all_classifications = all_classifications_query.all()
             classification = all_classifications[0]
             for i in all_classifications[1:]:
                 _db.delete(i)
 
-        classification.weight = weight
+        if classification.weight != weight:
```

Comment:
Why can't we go straight to the assignment here? Is this an optimization for the database? Something to appease `mypy`? Might be worth a comment (or just get smarter code reviewers 😆).

Reply:
An assignment would work fine here. I added this if statement because `weight` should already be set most of the time, and I didn't want to have SQLAlchemy flush it out to the database if its already set. I'm not sure if its a necessary optimization or not.

Reply:
did some reading of sqlalchemy docs, and apparently this isn't necessary, the ORM will handle internally, so i reverted this change